### PR TITLE
Show unlisted channels on Remote Import's Available Channels Page

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Adds some hacks to show unlisted channels from Kolibri Studio at the top of the Available Channels Page
1. Updates UI to show "open lock" icon around unlisted channels. (In import modes only; lock is not shown in export mode)

screen shot 2017-11-30 at 10 35 09 am

screen shot 2017-11-30 at 10 35 14 am

#### Happy path testing
1. Publish an unlisted channel
1. Import the channel to your Kolibri server with or without resources (the channel will not appear without resources).
1. Check that the unlisted channel appears as described above.

#### Un happy path testing
1. Somehow get a channel on a USB drive that is not on Kolibri Studio (neither listed/unlisted)
1. Install it to Kolibri
1. Check that the mystery channel does not appear the way unlisted channels do above
…

### References

Addresses #2720 

…

----

### Contributor Checklist

- [x] PR has the correct target milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, it has been assigned or requests review from someone and labeled as 'needs review'
- [ ] PR has been fully tested manually
- [ ] Documentation is updated
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Link to diff of internal dependency change is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] PR does not introduce [accessibility regressions](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing)
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

### CODE REVIEW

File: kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js [hunk 1]
```diff
@@ -48,3 +52,25 @@ export function showAvailableChannelsPage(store) {
...
+/**
+ * Makes a request to Kolibri Studio to get info on unlisted channels. And append
+ * them to the public channels.
+ *
+ * @param {Array<Channel>} publicChannels - the list of publich channels, which will not be queried
+ * @returns {Promise<Array<Channel>>}
+ */
+export function getAllRemoteChannels(store, publicChannels) {
+  const installedChannels = installedChannelList(store.state);
+  const potentiallyUnlisted = differenceBy(installedChannels, publicChannels, 'id').filter(
```
Comment (rtibbles) [inline kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: A little bit confused why this is necessary, when the data we get back from content curation should have a `public` flag on it to indicate that the channel is public, is this not being passed through from Kolibri to the frontend? https://github.com/fle-internal/content-curation/blob/develop/contentcuration/contentcuration/serializers.py#L708

Comment (jonboiser) [reply kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: Kolibri Studio's Channel's API only returns public channels (i.e. where `public: true`). The only API available to check on unlisted channels (`public: false`) is the lookup endpoint.

Don't know if this is possible, but if there were a way query the Channel's API to get both the public and unlisted-but-intalled-on-my-device channels, that would be nice.

Comment (rtibbles) [reply kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: Seems like this data is being sent up to the front end also: https://github.com/learningequality/kolibri/blob/release-v0.7.x/kolibri/content/api.py#L401

Comment (jonboiser) [reply kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: Yeah, that's how the publicChannels are represented. But there is no obvious way to get the very specific group of unlisted-but-installed-on-my-device channels. If there were, then we could get all such channels in one go, and I would just read off the `public` flag.

Comment (jonboiser) [reply kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: The main issue is that the ChannelMetadata does not have this `public` attribute.

Comment (rtibbles) [reply kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js:67]: Ah, ok - got it now. OK, let's leave this approach as is then, until we can make the API work more neatly for the frontend.

**Comment by indirectlylit** [comment]
looks like there's a conflict now

**Pull request merged**